apps/champions/lib/champions/users.ex

```diff
@@ -88,19 +83,37 @@ defmodule Champions.Users do
     Currencies.add_currency(user.id, Currencies.get_currency_by_name!("Gems").id, 500)
   end
 
+  defp add_campaigns_progress(user) do
+    campaigns = GameBackend.Campaigns.get_campaigns()
+
+    Enum.each(campaigns, fn campaign ->
+      if campaign.campaign_number == 1,
+        do:
+          GameBackend.Campaigns.insert_campaign_progress(%{
+            game_id: Utils.game_id(),
+            user_id: user.id,
+            campaign_id: campaign.id,
+            level_id: campaign.levels |> Enum.sort_by(& &1.level_number) |> hd() |> Map.get(:id)
+          })
+    end)
+  end
+
   @doc """
   Adds the given experience to a user. If the user were to have enough resulting experience to level up,
   it is performed automatically.
   """
   def add_experience(user_id, experience) do
-    user = get_user(user_id)
-    new_experience = user.experience + experience
+    case get_user(user_id) do
+      {:ok, user} ->
+        new_experience = user.experience + experience
```

Comment:
We can replace this case pattern-matching the get_user/1 returning value:
```suggestion
    {:ok, user} = get_user(user_id)
    new_experience = user.experience + experience
```

Reply:
That would make our application raise if the user cannot be found. Isn't it better to handle these expected errors gracefully?

Reply:
Yes, you're right.